ouroboros-network/src/Ouroboros/Network/PeerSelection/Governor/Types.hs

```diff
@@ -565,49 +577,335 @@ toPublicState PeerSelectionState { knownPeers
+      numberOfEstablishedBootstrapPeers          = Set.size   establishedBootstrapPeersSet,
+      numberOfWarmBootstrapPeersDemotions        = Set.size $ establishedBootstrapPeersSet `Set.intersection` inProgressDemoteWarm,
+      numberOfWarmBootstrapPeersPromotions       = Set.size $ establishedBootstrapPeersSet `Set.intersection` inProgressPromoteWarm,
+      numberOfActiveBootstrapPeers               = Set.size   activeBootstrapPeersSet,
+      numberOfActiveBootstrapPeersDemotions      = Set.size $ activeBootstrapPeersSet `Set.intersection` inProgressDemoteHot,
+
+      numberOfKnownLocalRootPeers                = Set.size   knownLocalRootPeersSet,
+      numberOfEstablishedLocalRootPeers          = Set.size $ establishedLocalRootsPeersSet,
+      numberOfWarmLocalRootPeersPromotions       = Set.size $ establishedLocalRootsPeersSet `Set.intersection` inProgressPromoteWarm,
+      numberOfActiveLocalRootPeers               = Set.size   activeLocalRootsPeersSet,
+      numberOfActiveLocalRootPeersDemotions      = Set.size $ activeLocalRootsPeersSet `Set.intersection` inProgressDemoteHot,
+
+      localRoots,
+
+      numberOfKnownSharedPeers                   = Set.size   knownSharedPeersSet,
+      numberOfColdSharedPeersPromotions          = Set.size $ knownSharedPeersSet `Set.intersection` inProgressPromoteCold,
+      numberOfEstablishedSharedPeers             = Set.size   establishedSharedPeersSet,
+      numberOfWarmSharedPeersDemotions           = Set.size $ establishedSharedPeersSet `Set.intersection` inProgressDemoteWarm,
+      numberOfWarmSharedPeersPromotions          = Set.size $ establishedSharedPeersSet `Set.intersection` inProgressPromoteWarm,
+      numberOfActiveSharedPeers                  = Set.size   activeSharedPeersSet,
+      numberOfActiveSharedPeersDemotions         = Set.size $ activeSharedPeersSet `Set.intersection` inProgressDemoteHot
     }
   where
-    knownPeersSet = KnownPeers.toSet (knownPeers st)
-    establishedPeersSet = EstablishedPeers.toSet (establishedPeers st)
-    coldPeersSet  = knownPeersSet Set.\\ establishedPeersSet
-    warmPeersSet  = establishedPeersSet Set.\\ activePeers
-    hotPeersSet   = activePeers
+    -- convention: only `{knwon,established,active}.*PeersSet` and
+    -- `inProgress{Promote,Demote}{Cold,Warm,Hot}` identifiers are used in the
+    -- calculations above ↑
+
+    -- common sets
+    knownSet       = KnownPeers.toSet knownPeers
+    establishedSet = EstablishedPeers.toSet establishedPeers
+    bigLedgerSet   = PublicRootPeers.getBigLedgerPeers publicRootPeers
+
+    -- non big ledger peers
+    knownPeersSet       = knownSet Set.\\ bigLedgerSet
+    establishedPeersSet = establishedSet Set.\\ bigLedgerSet
+    activePeersSet      = activePeers Set.\\ bigLedgerSet
+
+    -- big ledger peers
+    knownBigLedgerPeersSet       = knownSet `Set.intersection` bigLedgerSet
+    establishedBigLedgerPeersSet = establishedPeersSet `Set.intersection` bigLedgerSet
```

Comment:
establishedPeerSet -> establishedSet

Reply:
Right, otherwise `establishedBigLedgerPeersSet` will be empty.